File: loop_transformations/sources/fuse.2.f90
! @@name:       fuse.2
! @@type:       F-free
! @@operation:  compile
! @@expect:     success
! @@version:    omp_6.0

subroutine func(k, n, m, A, B)
  implicit none
  integer  :: k,n,m
  real     :: A(0:n-1),B(k:m+k-1)
  integer        :: i,j
  real, external :: f,g
  !$omp fuse
    do i = 0,n-1
      A(i) = f(i)
    end do
    do j = k,k+(m-1)
      B(j) = g(j)
    end do
  !$omp end fuse
end subroutine

subroutine func_equivalent(k, n, m, A, B)
  implicit none
  integer  :: k,n,m
  real     :: A(0:n-1),B(k:m+k-1)
  integer        :: ij
  real, external :: f,g

  do ij=0,max(n,m)-1
    if (ij < n) A(ij)   = f(ij)
    if (ij < m) B(ij+k) = g(ij+k)
  end do
end subroutine

File: loop_transformations/sources/fuse.2.c
/*
* @@name:	fuse.2
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_6.0
*/
float f(int);
float g(int);

void func(int k, int n, int m, float* A, float* B)
{
  #pragma omp fuse
  {
    for (int i = 0; i < n; ++i)
      A[i] = f(i);
    for (int j = k; j < k+m; ++j)
      B[j] = g(j);
  }
}

void func_equivalent(int k, int n, int m, float* A, float* B)
{
  for (int ij = 0; ij < (m>n ? m : n); ++ij) {
    if (ij < n) A[ij] = f(ij);
    if (ij < m) B[k+ij] = g(k+ij);
  }
}
